includes/classes/InternalConnections/NetworkSiteConnection.php

```diff
@@ -278,41 +288,22 @@ public function pull( $items ) {
 				}
 			}
 
-			foreach ( $post_props as $key => $value ) {
-				$post_array[ $key ] = $value;
-			}
-
-			if ( ! empty( $item_array['post_id'] ) ) {
-				$post_array['ID'] = $item_array['post_id'];
-			} else {
-				unset( $post_array['ID'] );
-			}
-
-			if ( isset( $post_array['post_parent'] ) ) {
-				unset( $post_array['post_parent'] );
-			}
-
-			if ( ! empty( $item_array['post_status'] ) ) {
-				$post_array['post_status'] = $item_array['post_status'];
-			}
-
 			add_filter( 'wp_insert_post_data', array( '\Distributor\InternalConnections\NetworkSiteConnection', 'maybe_set_modified_date' ), 10, 2 );
 
 			// Filter documented in includes/classes/ExternalConnections/WordPressExternalConnection.php
 			$new_post_args = Utils\post_args_allow_list( apply_filters( 'dt_pull_post_args', $post_array, $item_array['remote_post_id'], $post, $this ) );
-			$new_post_id   = wp_insert_post( wp_slash( $new_post_args ) );
-
+			if ( $update ) {
```

Comment:
@peterwilsoncc  `wp_insert_post` handles post update and insert. Can we use it? instead of `if...else`

Reply:
> `wp_insert_post` handles post update and insert. Can we use it? instead of `if...else`

[`wp_update_post`](https://github.com/WordPress/wordpress-develop/blob/4ad3cad2e0d9369d96f04837babe6270a1423487/src/wp-includes/post.php#L4752-L4818) includes a number of date, status and taxonomy checks that the previous code was either missing or doubling up, so I wanted to switch to the function to avoid the need to duplicate it.